SPECS/systemd/systemd.spec

```diff
@@ -209,7 +175,6 @@ fi
 %dir %{_sysconfdir}/modules-load.d
 %dir %{_sysconfdir}/binfmt.d
 %{_sysconfdir}/X11/xinit/xinitrc.d/50-systemd-user.sh
-%{_sysconfdir}/sysctl.d/50-security-hardening.conf
```

Comment:
Where is the recommended place to set the default hardening sysctls? I'd like to make sure we keep many of these settings

Reply:
these are already the default and so aren't actually 'hardening' anything, and don't need to be explicitly specified:
```
kernel.randomize_va_space=2
kernel.dmesg_restrict = 1
kernel.unprivileged_bpf_disabled = 1
net.ipv4.ip_forward = 0
net.ipv6.conf.all.forwarding = 0
```

it's debatable if this 'hardens' anything, since root can just set it to whatever else is desired, and it definitely reduces ease of debugging kernel issues (by disabling all sysrq operations and key sequences):
`kernel.sysrq=0`

this has nothing to do with 'hardening' and simply disables ipv6 autoconfiguration, which *at most* should be something that end users opt-in to (instead of end users having to opt-in to ipv6 autoconfiguration actually working):
`net.ipv6.conf.all.accept_ra = 0`

this actually '**de-hardens**' things by violating [rfc 5961](https://datatracker.ietf.org/doc/html/rfc5961):
`net.ipv4.tcp_challenge_ack_limit = 1073741823`

this is the only 'hardening' setting:
`kernel.kptr_restrict=1`

the `kernel.kptr_restrict=1` setting should go (along with any other 'identity-specific' config) into a `azure-linux-release-identity-*` package, similar to the `fedora-release-identity-*` packages. If it's a config that should go into *all* our images and/or install media, it could go into a `azure-linux-release-common` package, like the `fedora-release-common` package (we also should probably add a `99-default-disable.preset` into `azure-linux-release-common`, like the `fedora-release-common` package has).

For hardening-specific stuff we might want to have a `azure-linux-release-identity-hardened` or something that other more-specific release identity packages (that should have hardening) can require.

Reply:
These all make sense. I've booked some work to sanitize these settings, and I like the idea to what settings we really need to keep available in `azure-linux-release-identity-*` variants for making different variants & also having a `-common` package for settings we always want. Thanks for the very thorough explanation!

Reply:
https://microsoft.visualstudio.com/OS/_workitems/edit/47697769